test/unit/org/apache/cassandra/db/guardrails/GuardrailAllowFilteringTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GuardrailAllowFilteringTest extends GuardrailTester
+{
+    private boolean enableState;
+
+    @Before
+    public void setupTest()
+    {
+        createTable("CREATE TABLE %s (k int PRIMARY KEY, a int, b int)");
+        enableState = getGuardrial();
+    }
+
+    @After
+    public void teardownTest()
+    {
+        setGuardrail(enableState);
+    }
+
+    private void setGuardrail(boolean allowFilteringEnabled)
+    {
+        guardrails().setAllowFilteringEnabled(allowFilteringEnabled);
+    }
+
+    private boolean getGuardrial()
+    {
+        return guardrails().getAllowFilteringEnabled();
+    }
+
+    @Test
+    public void testAllowFilteringDisabled() throws Throwable
+    {
+        setGuardrail(false);
+        assertFails("SELECT * from %s WHERE a = 5 ALLOW FILTERING","Querying with ALLOW FILTERING is not allowed");
+    }
+
+    @Test
+    public void testAllowFilteringDisabedNotUsed() throws Throwable
```

Comment:
Nit: doesn't need the `throws Throwable` part if we are removing it from `CQLTester#execute`.

Reply:
Should I remove the `throws Throwable` from all of the methods that use it ? Or we can add that in another patch ?

Reply:
I was meaning this one, but the removal of the `throws` part on `CQLTester#execute` leaves thousands of methods with a similar unused `throws Throwable` part across unit tests. I think I would rather revert those changes on `CQLTester` and keep this patch focused on the new guardrail, leaving the removal of unneeded `throws Throwable` for a separate ticket. @josh-mckenzie wdyt?

Reply:
I think that the only remaining question is reverting the changes on `CQLTester`'s `throws Throwable` to do a proper cleanup in a follow up ticket, otherwise I'm +1 on the last changes.

Reply:
talked to @thingtwin1 and she will revert for now, can add in after 4.1 freeze